internal/pkg/skuba/kubernetes/node_version.go

```diff
@@ -179,29 +178,39 @@ func nodeVersioningInfo(client clientset.Interface, nodeName string) (NodeVersio
 
 	// find out the container image tags, depending on the role of the node
 	if IsControlPlane(nodeObject) {
-		apiServerTag, err := getPodContainerImageTag(client, metav1.NamespaceSystem, fmt.Sprintf("%s-%s", "kube-apiserver", nodeName))
+		// list all the pods
+		allPods, err := client.CoreV1().Pods(metav1.NamespaceSystem).List(metav1.ListOptions{})
+		// check for error
 		if err != nil {
-			return NodeVersionInfo{}, errors.Wrap(err, "could not retrieve apiserver pod")
+			return NodeVersionInfo{}, errors.Wrap(err, "could not retrieve pods")
 		}
-		controllerManagerTag, err := getPodContainerImageTag(client, metav1.NamespaceSystem, fmt.Sprintf("%s-%s", "kube-controller-manager", nodeName))
+		// check for empty pod list
+		if len(allPods.Items) == 0 {
+			return NodeVersionInfo{}, errors.New("list of pods is empty")
+		}
+		// check that the needed pods exist
+		apiserverPod, err := getPodFromPodList(allPods, "kube-apiserver-"+nodeName)
```

Comment:
Could be a matter of taste, but I prefer to use `Sprintf` when tweaking strings:

```suggestion
		apiserverPod, err := getPodFromPodList(allPods, fmt.Sprintf("kube-apiserver-%s", nodeName))
```

This applies to the other ones as well, if you prefer to keep the current way of appending the string it's fine for me :), just a matter of taste :)

Reply:
done!